src/track/serato/beatgrid.cpp

```diff
@@ -443,73 +443,59 @@ void SeratoBeatGrid::setBeats(BeatsPointer pBeats,
     auto pBeatsIterator = pBeats->findBeats(0, trackDurationSamples);
 
     // This might be null if the track doesn't contain any beats
-    if (!pBeatsIterator) {
+    if (!pBeatsIterator || !pBeatsIterator->hasNext()) {
         qWarning() << "Serato Beatgrid: No beats available, exporting empty beatgrid!";
         setTerminalMarker(nullptr);
         setNonTerminalMarkers({});
         return;
     }
 
+    const double beatgridFrameOffset = signalInfo.samples2framesFractional(pBeatsIterator->next());
     double currentBeatPositionFrames = 0;
     double previousBeatPositionFrames = 0;
     double previousDeltaFrames = -1;
-    int beatsSinceLastMarker = 0;
     QList<SeratoBeatGridNonTerminalMarkerPointer> nonTerminalMarkers;
+    {
+        const double positionSecs =
+                signalInfo.frames2secsFractional(
+                        currentBeatPositionFrames + beatgridFrameOffset) -
+                timingOffsetSecs;
+        nonTerminalMarkers.append(
+                std::make_shared<SeratoBeatGridNonTerminalMarker>(positionSecs, 0));
```

Comment:
Substract timingOffsetSecs from positionSecs when creating the marker, like we do for the inverse use case.